app/usedgoodreads/__main__.py

```diff
@@ -15,32 +15,59 @@ def index(key):
     cache = Book.get(key=key)
 
     if cache:
+        # Book is in database
         books = UsedBook.get(isbn=cache.isbn)
-
         if books:
+            # Used books are in database
             results = [{"title": v.title, "description": v.description,
                         "price": v.price, "link": v.link} for v in books]
 
             results.sort(key=lambda each: each["price"])
 
-            return render_template("index.html", isbn=cache.isbn, title=cache.title,
+            return render_template("index.html", isbn=cache.isbn,
+                                   title=cache.title,
                                    results=results)
-
-    ticket = Ticket(key)
-
-    if q.fetch_job(ticket.isbn_jid):
-        return render_template("index.html")
-
-    if q.fetch_job(ticket.used_books_jid):
-        return render_template("index.html", status="searching")  # TODO: Should return actual status
-
-    q.enqueue(resolve_goodreads_key, ticket, job_timeout=60 * 1,
-            ttl=60 * 60, job_id=ticket.isbn_jid)
-
-    q.enqueue(resolve_used_books, key, job_timeout=60 * 1,
-              ttl=60 * 60, job_id=ticket.used_books_jid, depends_on=ticket.isbn_jid)
-
-    return render_template("index.html")
+        else:
+            # Used books need to be fetched
+            ticket = Ticket(key)
+            job = q.fetch_job(ticket.used_books_jid)
+            if job:
+                # Fetch job already in q
+                if job.get_status() == "finished":
+                    # There have been no results at the end of the fetch job
+                    return render_template("index.html",
+                                           isbn=cache.isbn,
+                                           title=cache.title,
+                                           status="Used books: No results")
+                else:
+                    # fetch job still running or there was an error
+                    return render_template("index.html",
+                                           isbn=cache.isbn,
+                                           title=cache.title,
+                                           status=f"Used books: {job.get_status()}")
```

Comment:
Quick question, should we expose the rq status to the user here? Or just say something like "pending"?

Reply:
I don't think there is a problem with returning the actual status. They can be either `queued`, `started`, `deferred`, `finished`, and `failed`. Should be fine for the moment. Especially since we now update the site every 3 seconds the user should see some changing status information. 

We could also simply add a spinner? https://fontawesome.com/icons/spinner